### PR TITLE
station nuke(but not the nukeop nuke) activates delta alert

### PR DESCRIPTION
**Pull request by pedr69**
## What this does
image

newcop bombs remain silent, since they don't have a nanotrasen link
oh it also fixes a bug where the nuke was disarmed but kept blinking if you turned the safety on while it was armed

## Why it's good
what if instead of welderbombing toolstorage we nuclearbombed it?

## Changelog

:cl:
 * rscadd: nanotrasen nuclear bombs now trigger code delta when activated

### CODE REVIEW

**Comment by pedr69** [comment]
i feel like i'm going fucking insane, could have sworn that some sort of alarm used to play when malfs went delta, is it really just the "in1990space-cha-cha-analfisting-in1990" and then the silicon assault juke playlist?
is my localserb broken?

**Comment by west3436** [comment]
> i feel like i'm going fucking insane, could have sworn that some sort of alarm used to play when malfs went delta, is it really just the "in1990space-cha-cha-analfisting-in1990" and then the silicon assault juke playlist? is my localserb broken?

It plays that sound then starts the silicon playlist

**Comment by pedr69** [comment]
i guess i'm off my meds then, i VIVIDLY remember there being a "BWAAAAAAAAAAARP bebebebebeeep!"

File: code/modules/security levels/security levels.dm [hunk 1]
```diff
@@ -64,6 +64,7 @@
 					CC.post_status("alert", "redalert")*/
 
 			if(SEC_LEVEL_DELTA)
+				world << sound('sound/misc/imperial_alert.ogg')
```
Comment (gurfan) [inline code/modules/security levels/security levels.dm]: I think this is going to overlap with sound/AI/ai_malf.ogg, which plays when the AI does delta.

Comment (pedr69) [reply code/modules/security levels/security levels.dm:67]: it does
could try and delay the aimalf.ogg a couple seconds if you think it sounds bad

[malf.webm](https://user-images.githubusercontent.com/18052467/216388141-b1e169b0-9f75-47ad-98ca-132832890eaf.webm)

**Comment by SECBATON-GRIFFON** [comment]
> i feel like i'm going fucking insane, could have sworn that some sort of alarm used to play when malfs went delta, is it really just the "in1990space-cha-cha-analfisting-in1990" and then the silicon assault juke playlist? is my localserb broken?

its actually a glitched out version of that "in 1997, steve was chopping wood" thing

**Comment by gurfan** [comment]
> > i feel like i'm going fucking insane, could have sworn that some sort of alarm used to play when malfs went delta, is it really just the "in1990space-cha-cha-analfisting-in1990" and then the silicon assault juke playlist? is my localserb broken?
> 
> its actually a glitched out version of that "in 1997, steve was chopping wood" thing

how the fuck did you recognize this

**Comment by Eneocho** [comment]
ONLY the station nuke? Or does the ops nuke also trigger delta?

**Comment by pedr69** [comment]
>does the ops nuke also trigger delta
image

**Comment by Eneocho** [comment]
Oh I didn't notice that. My bad.
Though they DO have a nanotrasen link, they're stolen NT bombs, which is why you need the station's nuke disk to trigger.

File: code/modules/security levels/security levels.dm (hunk 1 above)
Comment (ValkyrieSkies) [reply code/modules/security levels/security levels.dm]: > it does could try and delay the aimalf.ogg a couple seconds if you think it sounds bad
> malf.webm

A counterpoint - if we're adding a generic nuclear detonation alert, we could arguably scrap the old AI sound for it, or repurpose it for something else. Having it just play a general warning could add a level of paranoia over whether someone has armed the nuke and there's not much time to find and disarm it, or if the AI has primed it and they should be rushing the AI instead.

Comment (gurfan) [reply code/modules/security levels/security levels.dm]: The glitchy AI sound is perfect and I definitely don't think it should be scrapped.

Comment (Eneocho) [reply code/modules/security levels/security levels.dm]: Keep the glitchy AI sound.

**Comment by pedr69** [comment]
could you explain the downdoots?

**Comment by Eneocho** [comment]
The alert sounds like ass to me and I prefer having the scuffed AI sounds.

**Comment by pedr69** [comment]
the scuffed AI sounds still played
i have however removed the siren

**Comment by Theophrastus1** [comment]
I just want the nostromo detonation alarm for Delta
[https://www.youtube.com/watch?v=EO9x0y5lqD0](url)

**Comment by pedr69** [comment]
bro your link

**Comment by ConfusedCactus** [comment]
I don't like this because going delta is something a malf AI does. I don't think we need to confuse people by making people think there's a malf AI when there isn't.

**Comment by pedr69** [comment]
I don't see how the code delta message would imply the AI is malfunctioning
when the AI is malf you get a "shutdown your AI" command alert and the silicon music starts playing

it just doesn't make sense to me that an AI hacking the nuke and overriding its safeties activates the alarm, but arming it normally doesn't

**Comment by Eneocho** [comment]
>I don't see how the code delta message would imply the AI is malfunctioning

Given that it only happens when the AI reveals the malfunction and the system detects the hacking override, it heavily implies the AI is malfunctioning. Plus there's an expectation now for the Delta = MALF AI, as it's been that way for a long while.
>it just doesn't make sense to me that an AI hacking the nuke and overriding its safeties activates the alarm, but arming it normally doesn't
The disk and code are the authorization to use the nuke, since they limit the nuke to being triggered locally and with a code no crewmember knows, thus no alarm is raised if the two are used normally. But the AI bypasses the disk and code, not authorized self destruction sequence, thus an alarm is raised.
Also, I mentioned it before, but Nuke Ops nukes ARE NT nukes, which is why they need to get that fukken disk before triggering it, so having the alarm blast with stationside nukes but not with nukeops nukes is not consistent.

**Comment by pedr69** [comment]
there is literally a `nt_aligned` var which nukeops bombs have set to 0
>Plus there's an expectation now for the Delta = MALF AI

that is true, but i find it silly that the station's emergency self destruct explosive just silently goes off instead of making an alarm, hence thie PR

**Comment by Kurfursten** [comment]
Lack of support

**Pull request closed without merging**